libs/blocks/global-navigation/features/profile/dropdown.js

```diff
@@ -161,10 +161,9 @@ class ProfileDropdown {
       </li>
     `;
 
-    // TODO consumers might want to execute their own logic before a sign out
-    // we might want to provide them a way to do so here
     signOutLink.addEventListener('click', (e) => {
       e.preventDefault();
+      window.dispatchEvent(new Event('feds:signOut'));
```

Comment:
This might still cause some race conditions, so in the future we might want to look into allowing consumers to define a method in the global namespace or their config that would be called here.